### PR TITLE
Add helper functions to find RSDP on PC BIOS

### PR DESCRIPTION
**Pull request by avdgrinten**
Every OS supporting the legacy BIOS likely needs such functions. Adding them to LAI makes the reimplementation of this code in every OS unnecessary.

### CODE REVIEW

File: helpers/pc-bios.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		if (lai_bios_calc_checksum(rsdp, sizeof(acpi_rsdp_t)))
+			continue;
+
+        // TODO: Handle XSDTs.
+		if(rsdp_addr)
+			*rsdp_addr = base + off;
+		if(rsdt_addr)
+			*rsdt_addr = rsdp->rsdt;
+		version = 1;
+        goto done;
```
Comment (thomtl) [inline helpers/pc-bios.c]: Is this goto really necessary, a `break` would seem better

File: helpers/pc-bios.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+done:
+    // TODO: Unmap the memory range.
+	return version;
+}
+
+int lai_bios_detect_rsdp(uintptr_t *rsdp_addr, uintptr_t *rsdt_addr) {
+	int version;
+	if((version = lai_bios_detect_rsdp_within(0x80000, 0x20000, rsdp_addr, rsdt_addr)))
```
Comment (thomtl) [inline helpers/pc-bios.c]: Technically the EBDA is not at `0x80000` for every PC, however the bios does specify the address of it in the BDA (see https://wiki.osdev.org/EBDA#BIOS_Data_Area_.28BDA.29 especially 0x40E). However to not open lai up to bios bugs some simply validity checking would have to be done

File: helpers/pc-bios.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+static uint8_t lai_bios_calc_checksum(void *ptr, size_t size) {
+	uint8_t sum = 0;
+	for (size_t i = 0; i < size; i++)
+		sum += ((uint8_t *)ptr)[i];
+	return sum;
+}
+
+int lai_bios_detect_rsdp_within(uintptr_t base, size_t length,
+		uintptr_t *rsdp_addr, uintptr_t *rsdt_addr) {
+    int version = 0;
```
Comment (thomtl) [inline helpers/pc-bios.c]: Indentation is off, at least on github

Comment (avdgrinten) [reply helpers/pc-bios.c]: I'll fix that.

File: helpers/pc-bios.c (hunk 1 above)
Comment (avdgrinten) [reply helpers/pc-bios.c]: This goto is necessary once we add a `laihost_unmap` function (which we should do).

File: helpers/pc-bios.c (hunk 2 above)
Comment (avdgrinten) [reply helpers/pc-bios.c]: You're right. ACPI does specify 0x40E though, so that should be more or less reliable, at least on hardware that supports ACPI.

**Comment by avdgrinten** [comment]
I fixed the EDBA and whitespace issues. The EBDA memory range was also off (I think I copied those from @qookei).

**Pull request merged**